src/sentry/api/serializers/models/notification_setting.py

```diff
@@ -35,32 +35,41 @@ def get_attrs(
         """
         type_option: Optional[NotificationSettingTypes] = kwargs.get("type")
 
-        # TODO(actorid) This needs to be reworked. `item_list` can either be
-        # a list of Team or User instances. Each item type requires different
-        # query logic for getting notification settings.
-        actor_mapping = {recipient.actor_id: recipient for recipient in item_list}
+        team_map = {}
+        user_map = {}
+        for recipient in item_list:
+            if isinstance(recipient, User):
+                user_map[recipient.id] = recipient
+            if isinstance(recipient, Team):
+                team_map[recipient.id] = recipient
```

Comment:
In addition to checking `User` and `Team` instance, do these also require checking `RpcUser` and `RpcTeam` respectively?

Reply:
I don't think we would encounter RpcTeam, but accounting for RpcUser is wise.